example/advection/advection_driver.cpp

```diff
@@ -49,10 +50,10 @@ TaskStatus UpdateContainer(MeshBlock *pmb, int stage,
   // const Real beta = stage_wghts[stage-1].beta;
   const Real beta = integrator->beta[stage - 1];
   const Real dt = integrator->dt;
-  Container<Real> &base = pmb->real_containers.Get();
-  Container<Real> &cin = pmb->real_containers.Get(stage_name[stage - 1]);
-  Container<Real> &cout = pmb->real_containers.Get(stage_name[stage]);
-  Container<Real> &dudt = pmb->real_containers.Get("dUdt");
+  auto &base = pmb->real_containers.Get();
+  auto &cin = pmb->real_containers.Get(stage_name[stage - 1]);
+  auto &cout = pmb->real_containers.Get(stage_name[stage]);
+  auto &dudt = pmb->real_containers.Get("dUdt");
```

Comment:
Yeah I don't know what best practices might be here.  To me, `std::shared_ptr` is just another object and I'd rather not just go around copying objects needlessly.  That may well be a flawed argument.  Also, it's possible these should really be something more like `const auto &` but what do I know.

Reply:
Yeah my logic is just that a shared pointer is just a reference-counted pointer. And a reference is just a pointer you don't have to de-reference. So this is like a pointer of a pointer. But your point is well taken. The reference semantics reduce copies of whatever is going on under the hood in shared pointer. I don't think it matters much. If you can add const, I like that.

Reply:
yeah, ok.  not going to add `const` for this one, but it might be good to try to remember in the future.

Reply:
Unnecessary copies of shared pointers has been shown to have performance implications in certain cases. I don't think it applies here.

Basically, a temporary copy of a shared pointer results in an atomic increment on copy and decrement when it goes out of scope. So, it can result in some unnecessary cache line invalidation. Probably doesn't matter much here.

Reply:
Ah. Cool. Thanks for the explanation, @AndrewGaspar !